data/templates/nginx/server.tpl.conf

```diff
@@ -20,41 +20,39 @@ server {
 }
 
 server {
-    # Disabling http2 for now as it's causing weird issues with curl
-    #listen 443 ssl http2;
-    #listen [::]:443 ssl http2;
-    listen 443 ssl;
-    listen [::]:443 ssl;
+    # curl is fixed
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+    #listen 443 ssl;
+    #listen [::]:443 ssl;
     server_name {{ domain }};
 
     ssl_certificate /etc/yunohost/certs/{{ domain }}/crt.pem;
     ssl_certificate_key /etc/yunohost/certs/{{ domain }}/key.pem;
-    ssl_session_timeout 5m;
-    ssl_session_cache shared:SSL:50m;
+    ssl_session_timeout 10m;
+    ssl_session_cache shared:SSL:10m;
+    ssl_session_tickets off;
+    
+    server_tokens off;
 
     # As suggested by Mozilla : https://wiki.mozilla.org/Security/Server_Side_TLS and https://en.wikipedia.org/wiki/Curve25519
     # (this doesn't work on jessie though ...?)
     # ssl_ecdh_curve secp521r1:secp384r1:prime256v1;
 
     # As suggested by https://cipherli.st/
-    ssl_ecdh_curve secp384r1;
+    #ssl_ecdh_curve secp384r1;
 
     ssl_prefer_server_ciphers on;
 
-    # Ciphers with intermediate compatibility
-    # https://mozilla.github.io/server-side-tls/ssl-config-generator/?server=nginx-1.6.2&openssl=1.0.1t&hsts=yes&profile=intermediate
-    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
-    ssl_ciphers 'ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS';
-
     # Ciphers with modern compatibility
     # https://mozilla.github.io/server-side-tls/ssl-config-generator/?server=nginx-1.6.2&openssl=1.0.1t&hsts=yes&profile=modern
     # Uncomment the following to use modern ciphers, but remove compatibility with some old clients (android < 5.0, Internet Explorer < 10, ...)
-    #ssl_protocols TLSv1.2;
-    #ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256';
+    ssl_protocols TLSv1.2;
+    ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
 
     # Uncomment the following directive after DH generation
     # > openssl dhparam -out /etc/ssl/private/dh2048.pem -outform PEM -2 2048
-    #ssl_dhparam /etc/ssl/private/dh2048.pem;
+    ssl_dhparam /etc/ssl/private/dh2048.pem;
```

Comment:
non. La clef ne peut pas être générée sur de petite architecture, ou prendra trop de temps. De toute façon cela nécessite de la créer au cours de la postinstall. La génération de cette clef pourrait prendre plusieurs heures pour un rpi ou orangepi par exemple.

Reply:
D'accord pour le RPI je peut le comprendre !

Reply:
I think the best way for that is to enable it only if the file `/etc/ssl/private/dh2048.pem`.

Reply:
Ajouter ça c'est B à A+ donc c'est très important tu génère la clé une fois c'est fini !

Reply:
Maybe we can write a message to the user at the end of postinstall to inform that it's recommended to create this file and how to create it.

Reply:
Ou alors on pourrait automatiser ça vous voulez une version secure ou light ?
Selon le choix cela modifie le fichier ?
La sécurité devrait etre au plus haut est pour ceux qui ne veulent donc peut d'utilisateur pourront prendre la version light have un simple oui / non